released/packages/coq-coqprime-generator/coq-coqprime-generator.1.1.1/opam

```diff
@@ -0,0 +1,42 @@
+opam-version: "2.0"
+maintainer: "thery@sophia.inria.fr"
+homepage: "https://github.com/thery/coqprime"
+bug-reports: "https://github.com/thery/coqprime/issues"
+dev-repo: "git+https://github.com/thery/coqprime.git"
+license: "LGPL-2.1-only"
+authors: ["Laurent Théry"]
+build: [
+  # cd to a subfolder in opam is tricky, so just move gencertif to the root folder
+  [ "find" "gencertif" "-type" "f" "-exec" "mv" "{}" "." ";" ]
+  [ "autoreconf" "-i" "-s" ]
+  [ "./configure" "--prefix" prefix
+    # Optiosn for finding opam local gmp-ecm
+    "CPPFLAGS=-I%{prefix}%/include"
+    "LDFLAGS=-L%{lib}%"
+    # Options for homebrew on Intel silicon (overwriting the above)
+    "CPPFLAGS=-I%{prefix}%/include -I/opt/local/include" { os-distribution = "macports" & os = "macos" }
+    "LDFLAGS=-L%{lib}% -L/opt/local/lib" { os-distribution = "macports" & os = "macos" }
+    # Options for homebrew on Apple silicon (overwriting the above)
+    "CPPFLAGS=-I%{prefix}%/include -I/opt/homebrew/include" { os-distribution = "homebrew" & os = "macos" & arch = "arm64"}
+    "LDFLAGS=-L%{lib}% -L/opt/homebrew/lib" { os-distribution = "homebrew" & os = "macos" & arch = "arm64"}
+    # Options for Windows cygwin
+    "--build=%{arch}%-pc-cygwin" { os = "win32" & os-distribution = "cygwinports" }
+    "--host=%{arch}%-w64-mingw32" { os = "win32" & os-distribution = "cygwinports" }
+    "--target=%{arch}%-w64-mingw32" { os = "win32" & os-distribution = "cygwinports" }
+  ]
+  [ make "-j" "%{jobs}%" ]
+]
+install: [
+  [make "install"]
+]
+depends: [
+  "ocaml"
```

Comment:
@MSoegtropIMC surely you have some lower bound here? Does this really work with OCaml 4.02?

Reply:
I didn't try and find it difficult to try (short of making a test PR on the main repo). But the OCaml code is relatively simple, a single file with 450 lines requiring Str and Big_Int.

I guess most people will install this with Coq Platform with recent OCaml compilers and if someone really uses this with an old OCaml we can still change it.

Reply:
Btw.: num has "ocaml" {>= "4.06.0"} and I guess this is the main restriction.

Reply:
Not quite, since `num` version 0 is compatible with older versions of OCaml. It is only `num` version 1 that requires 4.06.0. Opam will install one or the other depending on your version of the compiler.

Reply:
It compiles with 4.02.3, and given how simple the code is, I am pretty sure it also compiles with OCaml 3, so I don't think we need a lower bound.

Reply:
Thanks @silene for testing! I guess I also should set up a few opam switches for older OCaml.